packages/itwin/property-grid/src/components/Header.tsx

```diff
@@ -6,32 +6,103 @@
 import "./Header.scss";
 import classnames from "classnames";
 import { SvgProgressBackwardCircular } from "@itwin/itwinui-icons-react";
-import { IconButton } from "@itwin/itwinui-react";
+import { IconButton, SearchBox  } from "@itwin/itwinui-react";
 import { PropertyGridManager } from "../PropertyGridManager";
 
-import type { PropsWithChildren } from "react";
+import { type PropsWithChildren, useEffect, useRef, useState } from "react";
+import type React from "react";
 
 /** @internal */
 export interface HeaderProps {
   className?: string;
   onBackButtonClick?: () => void;
+  title?: React.ReactNode;
+  headerTools?: React.ReactNode;
+  onSearchStringChange?: (searchInput: string) => void;
 }
 
-/** @internal */
-export function Header({ className, children, onBackButtonClick }: PropsWithChildren<HeaderProps>) {
-  return <div className={classnames("property-grid-react-panel-header", className)}>
-    {
-      onBackButtonClick
-        ? <IconButton
-          styleType="borderless"
-          onClick={onBackButtonClick}
-          title={PropertyGridManager.translate("header.back")}
-          className="property-grid-react-header-back-button"
-        >
-          <SvgProgressBackwardCircular />
-        </IconButton>
-        : null
+export interface DebouncedSearchBoxProps {
+  onSearchStringChange: (searchInput: string) => void;
+  onClose: () => void;
+  onOpen: () => void;
+  className: string;
+}
+
+function DebouncedSearchBox({ onSearchStringChange, onClose, onOpen, className }: DebouncedSearchBoxProps){
+  const [inputValue, setInputValue] = useState<string>("");
+  const onSearchStringChangeRef = useRef(onSearchStringChange);
+  onSearchStringChangeRef.current = onSearchStringChange;
```

Comment:
React docs says that you should not change ref value during render. https://react.dev/reference/react/useRef#referencing-a-value-with-a-ref

```suggestion
  useEffect(() => {
    onSearchStringChangeRef.current = onSearchStringChange;
  }, [onSearchStringChange]);
```